Code/Framework/AzCore/Platform/Common/WinAPI/AzCore/Module/DynamicModuleHandle_WinAPI.cpp

```diff
@@ -106,9 +106,14 @@ namespace AZ
             {
                 // If module already open, return false, it was not loaded.
                 alreadyLoaded = NULL != GetModuleHandleW(fileNameW);
-
-                // Note: Windows LoadLibrary has no concept of specifying that the module symbols are global or local
-                m_handle = LoadLibraryW(fileNameW);
+                // Check if we need to load the dll
+                bool loadDll = !CheckBitsAny(flags, LoadFlags::NoLoad);
+                // Even if the DLL is already loaded we still need to call LoadLibraryW to increment the ref count.
+                if (alreadyLoaded || loadDll)
+                {
+                    // Note: Windows LoadLibrary has no concept of specifying that the module symbols are global or local
+                    m_handle = LoadLibraryW(fileNameW);
+                }
```

Comment:
The logic here can be simplified to consistent with the Unix logic by usingh[GetModuleHandleExW](https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulehandleexw) which DOES increment the reference count.

```c++
if (m_handle = GetModuleHandleExW(fileNameW); m_handle == nullptr && !CheckBitsAny(flags, LoadFlags::NoLoad))
{
    // Note: Windows LoadLibrary has no concept of specifying that the module symbols are global or local
    m_handle = LoadLibraryW(fileNameW);
}